src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,149 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private Object[] elementData;
+    private int size = 0;
+
+    public ArrayList() {
+        elementData = DEFAULTCAPACITY_EMPTY_ELEMENTDATA;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        rangeCheckForAdd(index);
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        System.arraycopy(this.elementData, index, this.elementData, index + 1, size - index);
+        this.elementData[index] = value;
+        size++;
+    }
+
+    private void rangeCheckForAdd(int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private void rangeCheck(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
 
+    private String outOfBoundsMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
     }
 
     @Override
     public void addAll(List<T> list) {
+        Object[] a = list.toArray();
+        int numNew = a.length;
+        if (numNew == 0) {
+            return;
+        }
+        Object[] elementData;
+        final int s;
+        if (numNew > (elementData = this.elementData).length - (s = size)) {
+            elementData = grow(s + numNew);
+        }
+        System.arraycopy(a, 0, elementData, s, numNew);
+        size = s + numNew;
+    }
```

Comment:
Try using add() method here. It has similar logic and will automatically grow an array if it will needed. Add every element of list object using loop.